plugins/package-managers/python/src/main/kotlin/Uv.kt

```diff
@@ -0,0 +1,138 @@
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.plugins.packagemanagers.python
+
+import java.io.File
+
+import org.apache.logging.log4j.kotlin.logger
+
+import org.ossreviewtoolkit.analyzer.PackageManager
+import org.ossreviewtoolkit.analyzer.PackageManagerFactory
+import org.ossreviewtoolkit.downloader.VersionControlSystem
+import org.ossreviewtoolkit.model.Identifier
+import org.ossreviewtoolkit.model.Project
+import org.ossreviewtoolkit.model.ProjectAnalyzerResult
+import org.ossreviewtoolkit.model.Scope
+import org.ossreviewtoolkit.model.config.AnalyzerConfiguration
+import org.ossreviewtoolkit.model.config.Excludes
+import org.ossreviewtoolkit.plugins.api.OrtPlugin
+import org.ossreviewtoolkit.plugins.api.PluginDescriptor
+import org.ossreviewtoolkit.plugins.packagemanagers.python.utils.PythonInspector
+import org.ossreviewtoolkit.plugins.packagemanagers.python.utils.getPythonVersion
+import org.ossreviewtoolkit.plugins.packagemanagers.python.utils.getPythonVersionConstraint
+import org.ossreviewtoolkit.plugins.packagemanagers.python.utils.toOrtPackages
+import org.ossreviewtoolkit.plugins.packagemanagers.python.utils.toPackageReferences
+import org.ossreviewtoolkit.utils.common.CommandLineTool
+import org.ossreviewtoolkit.utils.common.safeDeleteRecursively
+import org.ossreviewtoolkit.utils.ort.createOrtTempFile
+
+internal object UvCommand : CommandLineTool {
+    override fun command(workingDir: File?) = "uv"
+
+    override fun transformVersion(output: String) = output.substringAfter("version ").removeSuffix(")")
+}
+
+/**
+ * [Uv](https://github.com/astral-sh/uv) package manager for Python.
+ */
+@OrtPlugin(
+    displayName = "Uv",
+    description = "An extremely fast Python package and project manager.",
+    factory = PackageManagerFactory::class
+)
+class Uv(override val descriptor: PluginDescriptor = UvFactory.descriptor, private val config: PipConfig) :
+    PackageManager("Uv") {
+    companion object {
+        /**
+         * The name of the build system requirements and information file used by modern Python packages.
+         */
+        internal const val PYPROJECT_FILENAME = "pyproject.toml"
+    }
+
+    override val globsForDefinitionFiles = listOf("uv.lock")
+
+    override fun resolveDependencies(
+        analysisRoot: File,
+        definitionFile: File,
+        excludes: Excludes,
+        analyzerConfig: AnalyzerConfiguration,
+        labels: Map<String, String>
+    ): List<ProjectAnalyzerResult> {
+        val scopeName = parseScopeNamesFromPyproject(definitionFile.resolveSibling(PYPROJECT_FILENAME))
+        val resultsForScopeName = scopeName.associateWith { inspectLockfile(definitionFile) }
+
+        val packages = resultsForScopeName
+            .flatMap { (_, results) -> results.packages }
+            .toOrtPackages()
+            .distinctBy { it.id }
+            .toSet()
```

Comment:
This also looks unnecessary. As I realize you've copied this from the `Poetry` implementation, I'm removing that as part of https://github.com/oss-review-toolkit/ort/pull/10051 as well.